src/sjcl-ecdsa-signdeterministic.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+var sjcl = require('sjcl');
+/* eslint-disable new-cap */
+
+sjcl.ecc.ecdsa.secretKey.prototype.generateK =
+function(hash, hashObject) {
+
+  var curve = this._curve;
+  var qlen = sjcl.bitArray.bitLength(curve.r.toBits());
+
+  /* Utility functions */
+  /* used to generate k and v */
+  function repeat(str, times) {
+    return (new Array(times + 1)).join(str);
+  }
+  function bits2int(bits) {
+
+    var blen = sjcl.bitArray.bitLength(bits);
+
+    if (blen > qlen) {
+      return sjcl.bn.fromBits(sjcl.bitArray.bitSlice(bits, 0, qlen));
+    } else if (qlen < blen) {
+      return sjcl.bn.fromBits(sjcl.bitArray.concat(
+        sjcl.codec.hex.toBits(repeat('0', Math.ceil((qlen - blen) / 4))),
+        curve.r.toBits()
+      ));
+    }
+    return sjcl.bn.fromBits(bits);
+  }
+  function int2octets(integer) {
+    var iModQ = integer.mulmod(new sjcl.bn(1), curve.r);
+
+    var rlen = 8 * Math.ceil(qlen / 8);
+    var ilen = sjcl.bitArray.bitLength(iModQ.toBits());
+
+    return sjcl.bitArray.concat(
+      sjcl.codec.hex.toBits(repeat('0', Math.ceil((rlen - ilen) / 4))),
+      iModQ.toBits()
+    );
+  }
+  function bits2octets(bits) {
+    return int2octets(bits2int(bits).mulmod(new sjcl.bn(1), curve.r));
+  }
+
+  function hmac(bits, key) {
+    var hmacK = new sjcl.misc.hmac(key, hashObject);
+    return hmacK.encrypt(bits);
+  }
+
+  var hlen = sjcl.bitArray.bitLength(hash);
+  var x = sjcl.bn.fromBits(this.get());
+
+  var k = sjcl.codec.hex.toBits(repeat('00', Math.ceil(hlen / 8)));
```

Comment:
The issue I had using numbers was that it would ignore padding sometimes, or not convert to the same number of bits as I expected, which gave me a different result than the RFC tests